nvflare/tool/job/config/configer.py

```diff
@@ -103,11 +104,22 @@ def filter_config_name_and_values(
     return temp_results
 
 
+def _cast_type(key_index, cli_value):
+    """Casts cli_value to correct type.
+
+    Since build_reverse_order_index is using pyhocon, we need to do the same here.
+    """
+    if key_index.value is None:
+        return cli_value
+
+    new_value = ConfigFactory.parse_string(f"{key_index.key}={cli_value}")[key_index.key]
```

Comment:
Just to double check, if cli_value is "false" rather than "False" does this still convert correctly to False?

Reply:
thanks, let me try adding some unit test cases here as well.